### DISCUSSION TITLE
origin/branch not delete

### DISCUSSION THREAD

**Discussion by gdluna** [community/community#30599] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi! I delete from VSC a branch (branch_name), but origin/branch_name (remote branch) still showing. How can I delete it definitively, so this not showing? Thanks!
![vscbranch](https://user-images.githubusercontent.com/37933645/186282670-f6f788b2-7b39-40ba-9bfa-99fdfb17b79f.png)

**Comment by airtower-luna** [comment, marked as answer]
Deleting your a branch in your local repository doesn't touch the remote branch at all, to get rid of it you need to delete it in the remote repository too. You can do that [via `git push`](https://git-scm.com/docs/git-push#_examples) (second to last example), or possibly by other means depending on where your remote repository is (e.g. the branches list of a GitHub repository).

Git won't automatically delete a remote branch though (it's a precaution, you might still want it). To get rid of refs deleted on the remote use [`git remote prune origin`](https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-empruneem) (for the `origin` remote).

**Reply by gdluna** [reply]
Hi airtower-luna, , thank you very much for the explanation!
